src/components/list/SearchableList.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { FC, useCallback, useEffect, useState } from 'react';
+import Fuse, { FuseResult } from 'fuse.js';
+
+import { SearchField, SearchFieldOptions } from '../input/SearchField';
+import { Flex } from '../flex';
+
+import { ListItemType, SearchableListItem } from './SearchableListItem';
+
+import { List } from './index';
+
+export type ListItemWithChildren = ListItemType & {
+  children?: ListItemType[];
+};
+
+export type Props = {
+  listItems: ListItemWithChildren[];
+  NoResults?: FC;
+  EmptyQueryPlaceholder?: FC;
+  listAriaLabel?: string;
+  searchFieldOptions?: SearchFieldOptions;
+  listOptions?: { columns?: number; childrenSpacing?: 'md' | '2xl' };
+  listRef?: React.RefObject<HTMLUListElement | null>;
+};
+
+type FuseSearch = Fuse<ListItemWithChildren> & {
+  options: typeof fuseOptions;
+};
+
+const fuseOptions = {
+  keys: ['label', 'children.label'],
+  includeMatches: true,
+  threshold: 0,
+  ignoreLocation: true,
+  minMatchCharLength: 1,
+  shouldSort: false,
+};
+
+const mapItemsFromSearchResult = (
+  searchResults: FuseResult<ListItemWithChildren>[],
+): ListItemWithChildren[] => {
+  return searchResults.map(({ matches = [], item }) => {
+    const parentMatch = matches.find((match) => match.key === 'label');
+    const parentHighlightIndices = parentMatch ? parentMatch.indices : [];
+    const childMatches = item.children
+      ? matches.filter((match) => match.key === 'children.label')
+      : [];
+
+    if (
+      !item.children ||
+      item.children.length === 0 ||
+      childMatches.length === 0
+    ) {
+      return {
+        ...item,
+        highlightIndices: parentHighlightIndices,
+      };
+    }
+
+    const itemChildren = item.children || [];
+    const children = childMatches.reduce<ListItemType[]>(
+      (accumulator, childMatch) => {
+        if (childMatch.refIndex === undefined) {
+          return accumulator;
+        }
+
+        const child = itemChildren[childMatch.refIndex];
+        accumulator.push({
+          ...child,
+          highlightIndices: childMatch.indices,
+        });
+        return accumulator;
+      },
+      [],
+    );
+
+    return {
+      ...item,
+      ...(children ? { children } : {}),
+      highlightIndices: parentMatch ? parentMatch.indices : [],
+    };
+  });
+};
+
+const empty = () => null;
+
+const search = ({ query, fuse }: { query: string; fuse: FuseSearch }) => {
+  fuse.options.minMatchCharLength = query.length || 1;
+  return fuse.search(query);
+};
+
+const getFuseInstance = (listItems: ListItemWithChildren[]) => {
+  // Not casting the instance will result in the options not being typed
+  return new Fuse(listItems, fuseOptions) as unknown as FuseSearch;
+};
+
+export const SearchableList: FC<Props> = ({
+  listItems,
+  NoResults = empty,
+  EmptyQueryPlaceholder = empty,
+  listAriaLabel = 'searchable list',
+  searchFieldOptions = {},
+  listOptions = { columns: 1, childrenSpacing: 'md' },
+  listRef,
+}) => {
```

Comment:
i see that this is necessary to migrate for the header but please create a ticket to make it visible. I think it was missed when creating the initial set of stories for the migration